test/v3/services/RetrieveBalanceAndTransactionsServiceSpec.scala

```diff
@@ -0,0 +1,121 @@
+import v3.fixtures.retrieveBalanceAndTransactions.CodingDetailsFixture.codingDetails
+import v3.fixtures.retrieveBalanceAndTransactions.DocumentDetailsFixture.{documentDetails, documentDetailsWithoutDocDueDate}
+import v3.fixtures.retrieveBalanceAndTransactions.FinancialDetailsFixture.financialDetailsFull
+import v3.models.request.retrieveBalanceAndTransactions.RetrieveBalanceAndTransactionsRequestData
+import v3.models.response.retrieveBalanceAndTransactions._
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class RetrieveBalanceAndTransactionsServiceSpec extends ServiceSpec {
+
+  private val dateFrom           = LocalDate.parse("2018-08-13")
+  private val dateTo             = LocalDate.parse("2018-08-14")
+  private val includeStatistical = false
+
+  private val requestData: RetrieveBalanceAndTransactionsRequestData =
+    RetrieveBalanceAndTransactionsRequestData(
+      Nino("AA123456A"),
+      docNumber = Some("anId"),
+      Some(DateRange(dateFrom -> dateTo)),
+      onlyOpenItems = false,
+      includeLocks = false,
+      calculateAccruedInterest = false,
+      removePOA = false,
+      customerPaymentInformation = false,
+      includeEstimatedCharges = includeStatistical
+    )
+
+  val retrieveBalanceAndTransactionsResponse: RetrieveBalanceAndTransactionsResponse =
+    RetrieveBalanceAndTransactionsResponse(
+      balanceDetails = balanceDetails,
+      codingDetails = Some(List(codingDetails)),
+      documentDetails = Some(List(documentDetails, documentDetailsWithoutDocDueDate)),
+      financialDetails = Some(List(financialDetailsFull))
+    )
+
+  "RetrieveBalanceAndTransactionsService" when {
+    "the service call is successful" should {
+      "return the mapped result" in new Test {
+        MockRetrieveBalanceAndTransactionsConnector
+          .retrieveBalanceAndTransactions(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, retrieveBalanceAndTransactionsResponse))))
+
+        val result: ServiceOutcome[RetrieveBalanceAndTransactionsResponse] = await(service.retrieveBalanceAndTransactions(requestData))
+        result shouldBe Right(ResponseWrapper(correlationId, retrieveBalanceAndTransactionsResponse))
+      }
+    }
+
+    "map errors according to spec" when {
+      def serviceError(downstreamErrorCode: String, error: MtdError): Unit =
+        s"a $downstreamErrorCode error is returned from the service" in new Test {
+
+          MockRetrieveBalanceAndTransactionsConnector
+            .retrieveBalanceAndTransactions(requestData)
+            .returns(Future.successful(Left(ResponseWrapper(correlationId, DownstreamErrors.single(DownstreamErrorCode(downstreamErrorCode))))))
+
+          val result: ServiceOutcome[RetrieveBalanceAndTransactionsResponse] = await(service.retrieveBalanceAndTransactions(requestData))
+          result shouldBe Left(ErrorWrapper(correlationId, error))
+        }
+
+      val errors: Seq[(String, MtdError)] =
```

Comment:
```suggestion
      val errors =
```

Reply:
I quite like having the type there, it's used throughout